source/core/parameters.cc

```diff
@@ -750,6 +750,97 @@ namespace Parameters
     prm.leave_subsection();
   }
 
+  void
+  MeshBoxRefinement::declare_parameters(ParameterHandler &prm)
+    {
+        prm.enter_subsection("box refinement");
+        {
+            prm.declare_entry("type",
+                              "dealii",
+                              Patterns::Selection("gmsh|dealii|periodic_hills"),
+                              "Type of mesh "
+                              "Choices are <gmsh|dealii|periodic_hills>.");
+
+            prm.declare_entry("file name",
+                              "none",
+                              Patterns::FileName(),
+                              "GMSH file name");
+
+            prm.declare_entry("initial refinement",
+                              "0",
+                              Patterns::Integer(),
+                              "Initial refinement of the principal mesh");
+            prm.declare_entry("box initial refinement",
+                              "0",
+                              Patterns::Integer(),
+                              "Initial refinement of the box mesh");
```

Comment:
This is used in the case we are using a sphere or any curved mesh to define the box zone. In these cases if we want to have a smoother and rounder mesh refinement zone, we need to use a more refined mesh for the box. So we give the option to refine the mesh of the box. This is different from the refinement of the main mesh inside the box.

Reply:
Yeah but what I mean is that the parameter block you are using here, it could be a Mesh parameter block, it does not have to be a new type of block. You can use the same structure under a different variable name! Here you duplicated a bunch of code just for a synonym.